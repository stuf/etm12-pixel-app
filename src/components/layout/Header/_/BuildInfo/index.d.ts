import { FunctionComponent } from 'react';
import { BuildCommitEnv } from 'types';

export interface Props {
  env: BuildCommitEnv;
}

export interface Component extends FunctionComponent<Props> { }
