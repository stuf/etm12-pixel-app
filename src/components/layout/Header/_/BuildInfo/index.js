/* eslint no-unused-vars: [1, {"varsIgnorePattern": "[K|T]"}] */
import * as React from 'karet';
import * as U from 'karet.util';
import * as R from 'kefir.ramda';

import * as T from './index.d';

/**
 * @param {T.Props} props
 * @return {T.Component}
 */
function BuildInfo({ env = {} }) {
  return (
    <div className="flex-horizontal -vertical-center -justify-end pad -p-right">
      {U.unless(
        R.isEmpty(env),
        <div className="text -right -small">
          {env.REACT_APP_BUILD_VERSION} ({env.REACT_APP_BUILD_COMMIT}) – env:{' '}
          {env.REACT_APP_BUILD_ENV} – branch: {env.REACT_APP_BUILD_BRANCH}
        </div>,
      )}
    </div>
  );
}

export default BuildInfo;
