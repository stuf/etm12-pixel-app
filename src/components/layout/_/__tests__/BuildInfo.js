import * as React from 'karet';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BuildInfo from 'components/layout/_/BuildInfo';

const getEnv = () => ({
  REACT_APP_BUILD_VERSION: '1',
  REACT_APP_BUILD_COMMIT: '2',
  REACT_APP_BUILD_ENV: '3',
  REACT_APP_BUILD_BRANCH: '4',
});

it('renders correctly', () => {
  const wrapper = shallow(<BuildInfo />);
  const wrapper2 = shallow(<BuildInfo env={getEnv()} />);

  expect(toJson(wrapper)).toMatchSnapshot('without env');
  expect(toJson(wrapper2)).toMatchSnapshot('with env');
});
