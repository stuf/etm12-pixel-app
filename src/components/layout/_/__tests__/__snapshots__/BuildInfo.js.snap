// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: with env 1`] = `
<div
  className="flex-horizontal -vertical-center -justify-end pad -p-right"
>
  <div
    className="text -right -small"
  >
    1
     (
    2
    ) – env: 
    3
     
    – branch: 
    4
  </div>
</div>
`;

exports[`renders correctly: without env 1`] = `
<div
  className="flex-horizontal -vertical-center -justify-end pad -p-right"
/>
`;
